src/app/3d/qgs3dmapcanvasdockwidget.cpp

```diff
@@ -260,45 +260,53 @@ void Qgs3DMapCanvasDockWidget::configure()
 
   Qgs3DMapSettings *map = mCanvas->map();
   Qgs3DMapConfigWidget *w = new Qgs3DMapConfigWidget( map, mMainCanvas, &dlg );
-  QDialogButtonBox *buttons = new QDialogButtonBox( QDialogButtonBox::Ok | QDialogButtonBox::Cancel | QDialogButtonBox::Help, &dlg );
+  QDialogButtonBox *buttons = new QDialogButtonBox( QDialogButtonBox::Ok | QDialogButtonBox::Apply | QDialogButtonBox::Cancel | QDialogButtonBox::Help, &dlg );
   connect( buttons, &QDialogButtonBox::accepted, &dlg, &QDialog::accept );
   connect( buttons, &QDialogButtonBox::rejected, &dlg, &QDialog::reject );
+  connect( buttons, &QDialogButtonBox::clicked, &dlg, [ = ]( QAbstractButton * button )
+  {
+    if ( buttons->buttonRole( button ) == QDialogButtonBox::ApplyRole )
+      emit w->configChanged();
+  } );
   connect( buttons, &QDialogButtonBox::helpRequested, w, []() { QgsHelp::openHelp( QStringLiteral( "introduction/qgis_gui.html#scene-configuration" ) ); } );
 
   connect( w, &Qgs3DMapConfigWidget::isValidChanged, this, [ = ]( bool valid )
   {
     buttons->button( QDialogButtonBox::Ok )->setEnabled( valid );
   } );
 
-  QVBoxLayout *layout = new QVBoxLayout( &dlg );
-  layout->addWidget( w, 1 );
-  layout->addWidget( buttons );
-  if ( !dlg.exec() )
-    return;
+  connect( w, &Qgs3DMapConfigWidget::configChanged, this, [ = ]()
```

Comment:
Unless I'm missing something obvious, configChanged is never emitted if "ok" is clicked. So changes won't be saved if "apply" is not clicked before OK

Reply:
(to solve this it's probably cleanest to move this lambda code out to a private slot, and both connect it to the apply button  click and also call it after the dlg is executed.